module-4/backend/week_two.md

```diff
@@ -5,18 +5,59 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+in ES6 you have you have scoping uses classes.
+
 2. What's the difference between asynchronous and synchronous JavaScript? 
+Synchronous loading occurs when the browser must halt the rendering of the page in order to complete the execution of JavaScript code.
+Asynchronousis when JavaScript code is processed in parallel to the rest of the page content. 
+
 3. What are the four pillars of Object Oriented programming?
+Abstraction. 
+Encapsulation.
+Inheritance. 
+Polymorphism.
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+Refactoring?? not sure i understand
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+Look for functions larger than 10 lines of code, look for long conditionals, look for makint things into functions to be used later
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+A callback function is a function that is passed to another function as a parameter, and the callback function is called inside the other function. Youw really want to avoid using callbacks and use promises instead.
+
```

Comment:
Callbacks will still be unavoidable for things like array methods, *but* we do want to avoid them with our AJAX and use .then instead